src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -93,77 +94,140 @@ export const saveMetadata = createAsyncThunk<
         })),
     });
 
-    const chunkSize = 100;
-    for (let i = 0; i < metadata.length; i = i + chunkSize) {
-      const promises: Promise<void>[] = [];
-      for (let j = i; j < i + chunkSize && j < metadata.length; j++) {
-        const metadataItem: MetadataItem = metadata[j];
-        const index: string = metadataItem.rowKey;
-        if (
-          selectedMetadataKeys.includes(index) &&
-          metadataSaveDetails[index]?.saved !== true
-        ) {
-          const name: string = metadataItem[sampleNameColumn];
-          const metadataFields: FieldUpdate[] = Object.entries(metadataItem)
-            .filter(
-              ([key]) =>
-                headers.map((header) => header.name).includes(key) &&
-                key !== sampleNameColumn
-            )
-            .map(([key, value]) => ({
-              field: key,
-              value,
-              restriction: headers.filter((header) => header.name === key)[0]
-                .targetRestriction,
-            }));
-          const sampleId = metadataValidateDetails[index].foundSampleId;
-          if (sampleId) {
-            promises.push(
-              updateSample({
-                projectId,
-                sampleId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          } else {
-            promises.push(
-              createSample({
-                projectId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          }
-        }
-      }
-      await Promise.all(promises).then(() => {
-        dispatch(
-          setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+    const selectedSampleList = metadata.filter((metadataItem) => {
```

Comment:
Please add some comments to this file

Reply:
There's already comments in the file. Dd you mean add more comments within saveMetadata to describe what's happening? If so, please see 8845722f0d21be1856149625595c6d1fdc6c0c6b.

Reply:
Yup sorry I should've clarified. Thanks! :)